packages/plugin-npm-cli/sources/commands/npm/info.ts

```diff
@@ -17,130 +17,135 @@ declare module '@npm/types' {
 
-      If the optional field argument is provided, then only that part of the tree is returned.
+      If the \`-f,--fields\` option is set, it's a comma-separated list of fields which will be used to only display part of the package informations.
 
-      By default, \`yarn npm info\` will not return the \`dist\`, \`readme\`, and \`users\` fields, since they are often very long. To explicitly request those fields, use the \`-v,--verbose\` flag or the second argument: \`yarn npm info react readme\`
+      By default, this command won't return the \`dist\`, \`readme\`, and \`users\` fields, since they are often very long. To explicitly request those fields, explicitly list them with the \`--fields\` flag or request the output in JSON mode.
+
+      If the \`--json\` flag is set the output will follow a JSON-stream output also known as NDJSON (https://github.com/ndjson/ndjson-spec).
     `,
     examples: [[
       `Show all available information about react (except the \`dist\`, \`readme\`, and \`users\` fields)`,
       `yarn npm info react`,
-    ],[
-      `Show all available information about react, including the \`dist\`, \`readme\`, and \`users\` fields`,
-      `yarn npm info react --verbose`,
-    ],[
-      `Show all available information about react as valid JSON`,
+    ], [
+      `Show all available information about react as valid JSON (including the \`dist\`, \`readme\`, and \`users\` fields)`,
       `yarn npm info react --json`,
-    ],[
+    ], [
       `Show all available information about react 16.12.0`,
       `yarn npm info react@16.12.0`,
-    ],[
+    ], [
       `Show the description of react`,
-      `yarn npm info react description`,
-    ],[
+      `yarn npm info react --fields description`,
+    ], [
       `Show all available versions of react`,
-      `yarn npm info react versions`,
-    ],[
+      `yarn npm info react --fields versions`,
+    ], [
       `Show the readme of react`,
-      `yarn npm info react readme`,
+      `yarn npm info react --fields readme`,
+    ], [
+      `Show a few fields of react`,
+      `yarn npm info react --fields homepage,repository`,
     ]],
   });
 
   @Command.Path(`npm`, `info`)
   async execute() {
     const configuration = await Configuration.find(this.context.cwd, this.context.plugins);
+    const {project} = await Project.find(configuration, this.context.cwd);
 
-    /*
-     * Handle case when the package is the current workspace
-     */
-    if (this.package === `.`) {
-      const {project, workspace} = await Project.find(configuration, this.context.cwd);
-
-      if (!workspace)
-        throw new WorkspaceRequiredError(project.cwd, this.context.cwd);
+    const fields = typeof this.fields !== `undefined`
+      ? new Set(this.fields.split(/\s*,\s*/))
+      : null;
 
-      if (!workspace.manifest.name)
-        throw new UsageError(`Missing \`name\` field in ${path.join(workspace.cwd, `package.json`)}`);
+    if (fields !== null)
+      fields.add(`name`);
```

Comment:
Wouldn't it be better if the first field that is shown is always `name`?

Reply:
Sounds good to me 👍